HTTPServerProject.Tests/RequestTests.cs

```diff
@@ -0,0 +1,50 @@
+namespace HTTPServerProject.RequestTests;
+
+public class UnitTestsForRequests
```

Comment:
I think you've got an issue with the naming here - `UnitTestsForRequests` is testing the `Header` class.